influxql/ast.go

```diff
@@ -3437,20 +3449,22 @@ func TimeRangeAsEpochNano(expr Expr) (min, max int64) {
 
 // timeExprValue returns the time literal value of a "time == <TimeLiteral>" expression.
 // Returns zero time if the expression is not a time expression.
-func timeExprValue(ref Expr, lit Expr) time.Time {
+func timeExprValue(ref Expr, lit Expr) (t time.Time, err error) {
 	if ref, ok := ref.(*VarRef); ok && strings.ToLower(ref.Val) == "time" {
 		switch lit := lit.(type) {
 		case *TimeLiteral:
-			return lit.Val
+			t = lit.Val
 		case *DurationLiteral:
-			return time.Unix(0, int64(lit.Val)).UTC()
+			t = time.Unix(0, int64(lit.Val)).UTC()
 		case *NumberLiteral:
-			return time.Unix(0, int64(lit.Val)).UTC()
+			t = time.Unix(0, int64(lit.Val)).UTC()
 		case *IntegerLiteral:
-			return time.Unix(0, lit.Val).UTC()
+			t = time.Unix(0, lit.Val).UTC()
+		default:
+			err = fmt.Errorf("invalid operation: time and %T are not compatible", lit)
 		}
 	}
-	return time.Time{}
+	return
 }
```

Comment:
Named returns seem confusing here. Why not just use returns at each case instead of assigning `t` and falling through?